agent/engine/docker_container_engine.go

```diff
@@ -188,6 +208,7 @@ func NewDockerGoClient(clientFactory dockerclient.Factory, cfg *config.Config) (
 	}
 	return &dockerGoClient{
 		clientFactory:    clientFactory,
+		version:          version,
```

Comment:
I'm not sure why this is not necessary. In order to get the TaskEngine to use a different client version from the default, I think this field needs to be set explicitly. The current logic in the agent seems to always use the default client.